crates/subspace-farmer-components/src/file_ext.rs

```diff
@@ -1,20 +1,68 @@
 //! File extension trait
 
-use std::fs::File;
+use std::fs::{File, OpenOptions};
 use std::io::Result;
 
+pub trait OpenOptionsExt {
+    /// Advise OS/file system that file will use random access and read-ahead behavior is
+    /// undesirable, only has impact on Windows, for other operating systems see [`FileExt`]
+    fn advise_random_access(&mut self) -> &mut Self;
+
+    /// Advise OS/file system that file will use sequential access and read-ahead behavior is
+    /// desirable, only has impact on Windows, for other operating systems see [`FileExt`]
+    fn advise_sequential_access(&mut self) -> &mut Self;
+}
+
+impl OpenOptionsExt for OpenOptions {
+    #[cfg(target_os = "linux")]
+    fn advise_random_access(&mut self) -> &mut Self {
+        // Not supported
+        self
+    }
+
+    #[cfg(target_os = "macos")]
+    fn advise_random_access(&mut self) -> &mut Self {
+        // Not supported
+        self
+    }
+
+    #[cfg(not(any(target_os = "macos", target_os = "linux")))]
+    fn advise_random_access(&mut self) -> &mut Self {
+        use std::os::windows::fs::OpenOptionsExt;
+        self.custom_flags(winapi::um::winbase::FILE_FLAG_RANDOM_ACCESS)
+    }
+
+    #[cfg(target_os = "linux")]
+    fn advise_sequential_access(&mut self) -> &mut Self {
+        // Not supported
+        self
+    }
+
+    #[cfg(target_os = "macos")]
+    fn advise_sequential_access(&mut self) -> &mut Self {
+        // Not supported
+        self
+    }
+
+    #[cfg(not(any(target_os = "macos", target_os = "linux")))]
+    fn advise_sequential_access(&mut self) -> &mut Self {
+        use std::os::windows::fs::OpenOptionsExt;
```

Comment:
I would rather choose "windows" here and put a negative selector on "linux" option because it uses windows api explicitly. Are there reasons why choose these selectors?

Reply:
Not really, just copy-paste from before. I agree it makes sense to use Windows selector instead here